pkg/runner/step_context.go

```diff
@@ -79,33 +79,53 @@ func (sc *StepContext) Executor() common.Executor {
 	return common.NewErrorExecutor(fmt.Errorf("Unable to determine how to run job:%s step:%+v", rc.Run, step))
 }
 
-func (sc *StepContext) setupEnv() common.Executor {
+func (sc *StepContext) mergeEnv() map[string]string {
 	rc := sc.RunContext
 	job := rc.Run.Job()
 	step := sc.Step
-	return func(ctx context.Context) error {
-		var env map[string]string
-		c := job.Container()
-		if c != nil {
-			env = mergeMaps(rc.GetEnv(), c.Env, step.GetEnv())
-		} else {
-			env = mergeMaps(rc.GetEnv(), step.GetEnv())
-		}
 
-		if (rc.ExtraPath != nil) && (len(rc.ExtraPath) > 0) {
-			s := append(rc.ExtraPath, `/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`)
-			env["PATH"] = strings.Join(s, `:`)
-		}
+	var env map[string]string
+	c := job.Container()
+	if c != nil {
+		env = mergeMaps(rc.GetEnv(), c.Env, step.GetEnv())
+	} else {
+		env = mergeMaps(rc.GetEnv(), step.GetEnv())
+	}
+
+	if (rc.ExtraPath != nil) && (len(rc.ExtraPath) > 0) {
+		s := append(rc.ExtraPath, os.Getenv("PATH"))
+		env["PATH"] = strings.Join(s, string(os.PathListSeparator))
+	}
```

Comment:
Could you revert those lines? Otherwise this will overwrite PR #462

Reply:
Done, sorry